tgext/utils/csrf.py

```diff
@@ -65,25 +108,17 @@ def put_token_in_form():
             return dict(token=request.csrf_token)
 
     In a view, you can render this token as a hidden field inside the form. The
-    hidden field must have a name ``_csrf_token``::
+    hidden field must have the name specified in configuration ``csrf.token_name``,
+    by default it is ``_csrf_token``::
 
         <form method="POST">
             <input type="hidden" name="_csrf_token" value="{{ token }}">
             ....
         </form>
     """
-    req = tg.request._current_obj()
-
-    secret, token_name, path, expires = _get_conf()
-    token = req.signed_cookie(token_name, secret=secret)
-    if token:
-        # We will reuse existing tokens
-        tg.response.signed_cookie(token_name, token, secret,
-                                  path=path, max_age=expires)
-        req.csrf_token = token.decode('utf8')
-    else:
-        _generate_csrf_token()
+    _generate_csrf_token()
     # Pages with CSRF tokens should not be cached
+    req = tg.request._current_obj()
```

Comment:
You can get rid of this line. `tg.request.headers` will work perfectly by itself and we would avoid leaking access to an internal method.